### PR TITLE
docs(config/configuration_file.md): document --config option

### LINKED ISSUE #1669: `--config` is not documented explicitly
### Type

- [ ] Content inaccurate
- [x] Content missing
- [ ] Typo

### URL

NA

### Description

See #1047

### PR DESCRIPTION
**Pull request by Stoiicc**
## Description
Documented the --config CLI option in the configuration documentation .

## Checklist

- [x] I have read the [contributing guidelines](https://commitizen-tools.github.io/commitizen/contributing/)

### Code Changes

- [ ] Add test cases to all the changes you introduce
- [ ] Run `uv run poe all` locally to ensure this change passes linter check and tests
- [ ] Manually test the changes:
  - [ ] Verify the feature/bug fix works as expected in real-world scenarios
  - [ ] Test edge cases and error conditions
  - [ ] Ensure backward compatibility is maintained
  - [ ] Document any manual testing steps performed
- [ ] Update the documentation for the changes

### Documentation Changes

- [x] Run `uv run poe doc` locally to ensure the documentation pages renders correctly
- [ ] Check and fix any broken links (internal or external) in the documentation

## Expected Behavior
Users can easily discover the --config option in the documentation.

## Additional Context
Fixes #1669

### CODE REVIEW

File: docs/config/configuration_file.md [hunk 1]
```diff
@@ -20,6 +20,14 @@ Configuration files are typically located in the root of your project directory.
 
 The first valid configuration file found will be used. If no configuration file is found, Commitizen will use its default settings.
 
+!!! note
+    Commitizen supports  explicitly specifying a configuration file using the `--config` option, which is useful when the configuration file is not located in the project root directory.
```
Comment (bearomorphism) [inline docs/config/configuration_file.md]: nit

```suggestion
    Commitizen supports explicitly specifying a configuration file using the `--config` option, which is useful when the configuration file is not located in the project root directory.
```

File: docs/config/configuration_file.md [hunk 2]
```diff
@@ -20,6 +20,14 @@ Configuration files are typically located in the root of your project directory.
...
+    When `--config` is provided, Commitizen will only load configuration from the specified file and will not search for configuration files using the default search order described above. If the specified configuration file does not exist, Commitizen will raise an error.
```
Comment (bearomorphism) [inline docs/config/configuration_file.md]: Specify which error is raised

File: docs/config/configuration_file.md [hunk 3]
```diff
@@ -20,6 +20,14 @@ Configuration files are typically located in the root of your project directory.
 
 The first valid configuration file found will be used. If no configuration file is found, Commitizen will use its default settings.
 
+!!! note
+    Commitizen supports explicitly specifying a configuration file using the `--config` option, which is useful when the configuration file is not located in the project root directory.
+    When `--config` is provided, Commitizen will only load configuration from the specified file and will not search for configuration files using the default search order described above. If the specified configuration file does not exist, Commitizen raises the `ConfigFileNotFound` error.If the specified configuration file exists but is empty, Commitizen raises the `ConfigFileIsEmpty` error.
```
Comment (bearomorphism) [inline docs/config/configuration_file.md]:
```suggestion
    When `--config` is provided, Commitizen will only load configuration from the specified file and will not search for configuration files using the default search order described above. If the specified configuration file does not exist, Commitizen raises the `ConfigFileNotFound` error. If the specified configuration file exists but is empty, Commitizen raises the `ConfigFileIsEmpty` error.
```
